Toggl.Storage.Realm/Queries/ProcessPushResultQuery.cs

```diff
@@ -0,0 +1,96 @@
+        {
+            Ensure.Argument.IsNotNull(realmProvider, nameof(response));
+            Ensure.Argument.IsNotNull(response, nameof(response));
+
+            this.realmProvider = realmProvider;
+            this.response = response;
+        }
+
+        public Unit Execute()
+        {
+            var realm = realmProvider();
+            using (var tx = realm.BeginWrite())
+            {
+                processSingletonResult<IPreferences, RealmPreferences>(response.Preferences, realm);
+                processSingletonResult<IUser, RealmUser>(response.User, realm);
+
+                foreach (var res in response.Workspaces)
+                    processResult<IWorkspace, RealmWorkspace>(res, realm);
+
+                foreach (var res in response.Tags)
+                    processResult<ITag, RealmTag>(res, realm);
+
+                foreach (var res in response.Clients)
+                    processResult<IClient, RealmClient>(res, realm);
+
+                foreach (var res in response.Tasks)
+                    processResult<ITask, RealmTask>(res, realm);
+
+                foreach (var res in response.Projects)
+                    processResult<IProject, RealmProject>(res, realm);
+
+                foreach (var res in response.TimeEntries)
+                    processResult<ITimeEntry, RealmTimeEntry>(res, realm);
+
+                tx.Commit();
+            }
+
+            return Unit.Default;
+        }
+
+        private void processSingletonResult<TEntity, TRealmEntity>(IActionResult<TEntity> result, Realms.Realm realm)
+            where TRealmEntity : Realms.RealmObject, IPushable
+        {
+            if (result?.Success == false && result is ErrorResult<TEntity> error)
+            {
+                var entity = realm.All<TRealmEntity>().Single();
+                entity.PushFailed(error.ErrorMessage.DefaultMessage);
+            }
+        }
+
+        private void processResult<TEntity, TRealmEntity>(IEntityActionResult<TEntity> entityResult, Realms.Realm realm)
+            where TRealmEntity : Realms.RealmObject, IPushable, IModifiableId, ISyncable<TEntity>
+        {
+            var entity = realm.All<TRealmEntity>().Where(e => e.Id == entityResult.Id).SingleOrDefault();
+
+            // It is possible that the entity was deleted locally since the push.
+            // In that case, simply ignore this result.
+            if (entity == null) return;
+
+            if (entityResult.Result.Success == false && entityResult.Result is ErrorResult<TEntity> error)
+            {
+                switch (error.ErrorMessage.Code)
+                {
+                    case ErrorCode.Skipped:
+                        break;
+                    default:
+                        entity.PushFailed(error.ErrorMessage.DefaultMessage);
+                        break;
+                }
+            }
+            else if (entityResult is CreateActionResult<TEntity> create
+                && create.Result.Success == true
```

Comment:
```suggestion
                && create.Result.Success
```

Seriously, had you named the property `IsSuccess`, this weird redundancy would be more obvious 😄